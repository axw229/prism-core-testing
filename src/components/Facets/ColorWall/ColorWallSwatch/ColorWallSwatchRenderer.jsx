// @flow
import React from 'react'

import { CLASS_NAMES } from './shared'

import './ColorWallSwatch.scss'

type Props = {
  color: string,
  disabled?: boolean,
  focus?: boolean
}

const ColorWallSwatchRenderer = ({ color, disabled, focus }: Props) => {
  return (
    <div className={CLASS_NAMES.SWATCH}
      role='presentation'>
      <div
        className={`${CLASS_NAMES.BASE} ${CLASS_NAMES.BASE_DISABLED} ${focus ? CLASS_NAMES.BASE_FOCUS : ''}`}
        style={{ background: color }}>
        {disabled ? <div className={CLASS_NAMES.FLAG} /> : null}
      </div>
    </div>
  )
}

export default ColorWallSwatchRenderer
