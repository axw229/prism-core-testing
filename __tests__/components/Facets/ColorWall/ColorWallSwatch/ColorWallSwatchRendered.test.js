/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import ColorWallSwatchRenderer from 'src/components/Facets/ColorWall/ColorWallSwatch/ColorWallSwatchRenderer'

const getColorWallSwatchRenderer = (props) => {
  return shallow(<ColorWallSwatchRenderer {...props} />)
}

describe('snapshot test', () => {
  const ColorWallSwatchRenderer = getColorWallSwatchRenderer()
  it('ColorWallSwatchRenderer should match the Snapshot', () => {
    expect(ColorWallSwatchRenderer).toMatchSnapshot()
  })
})
