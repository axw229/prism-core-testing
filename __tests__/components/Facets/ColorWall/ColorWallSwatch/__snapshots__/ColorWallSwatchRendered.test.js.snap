// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test ColorWallSwatchRenderer should match the Snapshot 1`] = `
<div
  className="color-wall-swatches__swatch"
  role="presentation"
>
  <div
    className="color-swatch-inner "
    style={
      Object {
        "background": undefined,
      }
    }
  />
</div>
`;
